pkg/kubectl/apis/manifest/v1alpha1/types.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Package could be either a base or a overlay manifest.
+type Package struct {
+	metav1.TypeMeta `json:",inline" yaml:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	// Labels to add to all objects and selectors.
+	// These labels would also be used to form the selector for apply --prune
+	// Named differently than “labels” to avoid confusion with metadata for this object
+	ObjectLabels map[string]string `json:"objectLabels,omitempty" yaml:"objectLabels,omitempty"`
+
+	// Annotations to add to all objects.
+	ObjectAnnotations map[string]string `json:"objectAnnotations,omitempty" yaml:"objectAnnotations,omitempty"`
+
+	// Partial name that will prefix the name of the base resources.
+	NamePrefix string `json:"namePrefix,omitempty" yaml:"namePrefix,omitempty"`
+
+	// Description of the application package.
+	Description string `json:"description,omitempty" yaml:"description,omitempty"`
```

Comment:
Please put this kind of package "metadata" directly under ObjectMeta, above.

Reply:
Proposed order:
1. Description
2. Icon
3. Keywords
4. Home
5. Source
6. Maintainers (I could also imagine dropping this)
7. AppVersion

Then non-metadata fields.
1. NamePrefix
2. ObjectLabels
3. ObjectAnnotations
4. Bases
5. Overlays
6. ConfigMaps
7. Secrets
8. OwnPersistentVolumeClaims
9. Recursive
10. Prune

Reply:
Hmm. Actually, how about we put the package metadata into a separate resource?

The metadata would drive package searching and browsing, and fork/rebase upgrade workflows. It might need to be consumed by app registry and other tools.

The non-metadata fields would drive kubectl resource processing functionality.

Reply:
Yes, would be better to have 2 distinct resources between metadata and data used by the tools.